### PR TITLE
Docker perf and fixes

### PR DESCRIPTION
**Pull request by vjkoskela**
Add support for capturing performance traces using HPROF against a JVM running inside Docker. This allows integration test level performance testing in addition to the existing unit test level performance testing. 

The change also includes additional testing and refactoring to support and improve the code. While not perfect it's an incremental step in the right direction.

This project will need to be upgraded to another mechanism (in place of HPROF) in order to facilitate upgrades to JDK/JRE 11.

### CODE REVIEW

File: src/main/java/com/arpnetworking/test/junitbenchmarks/ContainerFileReader.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import java.io.Reader;
+import java.nio.file.Path;
+import javax.annotation.Nullable;
+
+/**
+ * Reader for files in Docker containers.
+ *
+ * @author Ville Koskela (ville dot koskela at inscopemetrics dot com)
+ */
+class ContainerFileReader extends BufferedReader {
```
[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/test/junitbenchmarks/ContainerFileReader.java]: Agreed. Done.

File: src/main/java/com/arpnetworking/test/junitbenchmarks/DockerJsonBenchmarkConsumer.java [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+        final Optional<Container> container = getContainer(_dockerClient, _targetImageName);
+        if (!container.isPresent()) {
+            throw new IOException("Cannot determine file size; container not found");
+        }
+        long fileSize = 0;
+        BufferedReader reader = null;
+        try {
+            reader = new ContainerFileReader(_dockerClient, container.get(), file);
+            while (reader.read() >= 0) {
+                ++fileSize;
```
[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/test/junitbenchmarks/DockerJsonBenchmarkConsumer.java]: Don't see how it matters; the `BufferedReader` that the `ContainerFileReader` is wrapped in ensures that this is an efficient operation.

[the remark this answers is not in the document]
Comment (vjkoskela) [reply src/main/java/com/arpnetworking/test/junitbenchmarks/DockerJsonBenchmarkConsumer.java]: Improved

**Comment by BrandonArp** [comment]
:ship::it:

**Pull request merged**